config/diff.go

```diff
@@ -513,3 +517,16 @@ func diffModule(left, right Module, diff *Diff) bool {
 	diff.Modified.Modules = append(diff.Modified.Modules, right)
 	return true
 }
+
+// diffLogCfg returns true if any part of the log config is different or if any
+// services or components have been updated.
+func diffLogCfg(left, right *Config, servicesDifferent, componentsDifferent bool) bool {
+	if !reflect.DeepEqual(left.LogConfig, right.LogConfig) {
+		return true
+	}
+	// If there was any change in services or components; attempt to update logger levels.
```

Comment:
I've added some basic log diffing here. It is lazy in the sense that I don't _actually_ look through `LogConfiguration` fields on services and components to determine if anything has changed (I treat any change in services or components as a potential change in logging behavior). I was lazy because I'm really only using the returned boolean here to avoid calling `s.registry.Update` on _every_ config update, and output a `Detected potential log configuration changes; updating logger levels` log.